### PR TITLE
ext_man: Port SOF_IPC_EXT_WINDOW to extended manifest

### PR DESCRIPTION
**Pull request by ktrzcinx**
This information known at build time does not need to be
provided in a mailbox.

https://github.com/thesofproject/linux/pull/2087

### CODE REVIEW

File: src/include/ipc/info.h [hunk 1]
```diff
@@ -107,7 +107,7 @@ struct sof_ipc_window_elem {
 struct sof_ipc_window {
 	struct sof_ipc_ext_data_hdr ext_hdr;
 	uint32_t num_windows;
-	struct sof_ipc_window_elem window[];
+	struct sof_ipc_window_elem window[7];
```
Comment (lgirdwood) [inline src/include/ipc/info.h:110]: Do you need to change this ?

File: src/include/kernel/abi.h [hunk 2]
```diff
@@ -29,7 +29,7 @@
 
 /** \brief SOF ABI version major, minor and patch numbers */
 #define SOF_ABI_MAJOR 3
-#define SOF_ABI_MINOR 16
+#define SOF_ABI_MINOR 17
```
Comment (lgirdwood) [inline src/include/kernel/abi.h:32]: ABI - do we have kernel PR ?

File: src/platform/baytrail/platform.c [hunk 3]
```diff
@@ -65,16 +66,20 @@ static const struct sof_ipc_fw_ready ready
...
+	_META_EXPAND(
 	.ext_hdr	= {
 		.hdr.cmd = SOF_IPC_FW_READY,
-		.hdr.size = sizeof(struct sof_ipc_window) +
-			sizeof(struct sof_ipc_window_elem) * NUM_BYT_WINDOWS,
+		.hdr.size = sizeof(struct sof_ipc_window),
 		.type	= SOF_IPC_EXT_WINDOW,
 	},
-	.num_windows	= NUM_BYT_WINDOWS,
+	.num_windows	= 6,
```
Comment (lgirdwood) [inline src/platform/baytrail/platform.c:82]: Why would we do this ?

File: src/include/ipc/info.h (hunk 1 above)
Comment (ktrzcinx) [reply src/include/ipc/info.h:110]: Used toolchain reject packing variable-size structures inside another one, even as last field - same situation as it was before for CC_VERSION

File: src/platform/baytrail/platform.c (hunk 3 above)
Comment (ktrzcinx) [reply src/platform/baytrail/platform.c:82]: After change, this macro was used only once - in this place

File: src/include/kernel/abi.h (hunk 2 above)
Comment (ktrzcinx) [reply src/include/kernel/abi.h:32]: https://github.com/thesofproject/linux/pull/2087

File: src/platform/baytrail/platform.c (hunk 3 above)
Comment (lgirdwood) [reply src/platform/baytrail/platform.c:82]: Doesn't matter how many times it used, it's a needless and unrelated change here.

Comment (ktrzcinx) [reply src/platform/baytrail/platform.c]: updated

**Comment by ktrzcinx** [comment]
I resolved conflicts

**Comment by lgirdwood** [comment]
Need to align MINOR ABI prior to merging.

**Comment by ktrzcinx** [comment]
@lgirdwood I can't see issue with ABI number, as I see I updated from 3.16.0 (master ABI) to 3.17.0, what is aligned with changes from https://github.com/thesofproject/linux/pull/2087

**Comment by lgirdwood** [comment]
@ktrzcinx sorry, I mean that @kv2019i and I have to align these between kernel and FW. We are probably good to merge, but I just need to align with @kv2019i that kernel is happy with the ABI numbers too.

**Comment by kv2019i** [comment]
@ktrzcinx wrote:
> @ktrzcinx sorry, I mean that @kv2019i and I have to align these between kernel and FW. We are probably good to merge, but I just need to align with @kv2019i that kernel is happy with the ABI numbers too.

Sorry a bit late. This is good to merge, patch approved on kernel side.

File: src/include/kernel/abi.h (hunk 2 above)
Comment (lgirdwood) [reply src/include/kernel/abi.h:32]: Will this PR work at MINOR 16 ? i.e. is there anything in 16 that this breaks ?

Comment (ktrzcinx) [reply src/include/kernel/abi.h:32]: This PR changes already used structure definition, and changes sof_ipc_window struct size calculation from kernel size - https://github.com/thesofproject/linux/pull/2087#discussion_r422852863 - so I bumped ABI. Even so, it's backward compatible.

**Pull request merged**